src/theory/theory_engine.cpp

```diff
@@ -392,6 +398,22 @@ void TheoryEngine::check(Theory::Effort effort) {
         d_relManager->beginRound();
       }
       d_tc->resetRound();
+
+      if (options::computePartitions() > 1 && options::partitionCheck() == options::CheckMode::FULL){
```

Comment:
Please check for `d_splitter != nullptr` here instead. This opens the possibility to change the condition in `finishInit()` at some point and then being out of sync, simply assuming `d_splitter != nullptr` when it is not. (Same below)